File: leetcode/medium/1214-two-sum-bsts/1214-two-sum-bsts.kt
///**
// * Example:
// * var ti = TreeNode(5)
// * var v = ti.`val`
// * Definition for a binary tree node.
class TreeNode(var `val`: Int) {
    var left: TreeNode? = null
    var right: TreeNode? = null
}

class Solution {
    fun bst(root: TreeNode?, map: MutableMap<Int, Int>) {
        if (root == null) {
            return
        }
        map[root.`val`] = 1
        bst(root.left, map)
        bst(root.right, map)
    }

    fun twoSumBSTs(root1: TreeNode?, root2: TreeNode?, target: Int): Boolean {
        var map1 = mutableMapOf<Int, Int>()
        var map2 = mutableMapOf<Int, Int>()

        bst(root1, map1)
        bst(root2, map2)

        for (key in map1.keys) {
            if (map2.containsKey(target - key)) {
                return true
            }
        }

        return false
    }
}

File: leetcode/medium/1214-two-sum-bsts/1214-two-sum-bsts.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def twoSumBSTs(self, root1: Optional[TreeNode], root2: Optional[TreeNode], target: int) -> bool:
        map1 = {}
        map2 = {}

        def bst(root, map):
            if root:
                map[root.val] = True
                bst(root.left, map)
                bst(root.right, map)

        bst(root1, map1)
        bst(root2, map2)

        for key in map1:
            if target - key in map2:
                return True

        return False
